controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -195,50 +195,50 @@ public void createStream(final String scopeName, final CreateStreamRequest creat
             LoggerHelpers.traceLeave(log, "createStream", traceId);
             return;
         }
 
-        StreamConfiguration streamConfiguration = ModelHelper.getCreateStreamConfig(createStreamRequest, scopeName);
-        controllerService.createStream(streamConfiguration, System.currentTimeMillis())
+        StreamConfiguration streamConfiguration = ModelHelper.getCreateStreamConfig(createStreamRequest);
+        controllerService.createStream(scopeName, streamName, streamConfiguration, System.currentTimeMillis())
                 .thenApply(streamStatus -> {
                     Response resp = null;
                     if (streamStatus.getStatus() == CreateStreamStatus.Status.SUCCESS) {
-                        log.info("Successfully created stream: {}/{}", scopeName, streamConfiguration.getStreamName());
+                        log.info("Successfully created stream: {}/{}", scopeName, streamName);
                         resp = Response.status(Status.CREATED).
-                                entity(ModelHelper.encodeStreamResponse(streamConfiguration)).build();
+                                entity(ModelHelper.encodeStreamResponse(scopeName, streamName, streamConfiguration)).build();
                     } else if (streamStatus.getStatus() == CreateStreamStatus.Status.STREAM_EXISTS) {
-                        log.warn("Stream already exists: {}/{}", scopeName, streamConfiguration.getStreamName());
+                        log.warn("Stream already exists: {}/{}", scopeName, streamName);
                         resp = Response.status(Status.CONFLICT).build();
                     } else if (streamStatus.getStatus() == CreateStreamStatus.Status.SCOPE_NOT_FOUND) {
                         log.warn("Scope not found: {}", scopeName);
                         resp = Response.status(Status.NOT_FOUND).build();
                     } else if (streamStatus.getStatus() == CreateStreamStatus.Status.INVALID_STREAM_NAME) {
-                        log.warn("Invalid stream name: {}", streamConfiguration.getStreamName());
+                        log.warn("Invalid stream name: {}", streamName);
                         resp = Response.status(Status.BAD_REQUEST).build();
                     } else {
-                        log.warn("createStream failed for : {}/{}", scopeName, streamConfiguration.getStreamName());
+                        log.warn("createStream failed for : {}/{}", scopeName, streamName);
                         resp = Response.status(Status.INTERNAL_SERVER_ERROR).build();
                     }
                     return resp;
                 }).exceptionally(exception -> {
-                    log.warn("createStream for {}/{} failed {}: ", scopeName, streamConfiguration.getStreamName(),
-                             exception);
-                    return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+                             log.warn("createStream for {}/{} failed {}: ", scopeName, streamName, exception);
```

Comment:
you are double-logging the exception. No need for the second `{}`

Reply:
I don't follow.

Reply:
I think @andreipaduroiu is referring to the fact that having a throwable at the end of the list is sufficient to have it logged. See it here for a discussion on it: https://github.com/pravega/pravega/issues/2736